scipy/cluster/hierarchy.py

```diff
@@ -3004,13 +3004,107 @@ def set_link_color_palette(palette):
     _link_line_colors = palette
 
 
+def _reorder_leaves(Z, leaves_order):
+    """"
+    Given a tree T, encoded az Z, builds a tree F
+    and its encoding new_Z s.t. leaves(F) == leaves_order
+    Variables:
+    ch[i] = children of node (n+i) in T
+    p - array of parents in tree T
+    height[i] is a value of Z[i][3] e.g. cost of creating node (n+i)
+    num_leaves[i] = number of leaves of a subtree rooted in a node i in T
+    """
```

Comment:
Add in `Parameters` and `Returns` such as in other functions. While this isn't a public function, these should still be documented as it is fairly complex function.